doc/tutorials/basic/progressive_layouts.md

```diff
@@ -0,0 +1,358 @@
+:::
+
+## Replace Content Progressively
+
+Run the code below:
+
+```{pyodide}
+import panel as pn
+from time import sleep
+import random
+
+pn.extension()
+
+OPTIONS = ["Wind Turbine", "Solar Panel", "Battery Storage"]
+
+def classify(image):
+    sleep(2)
+    return random.choice(OPTIONS)
+
+
+run = pn.widgets.Button(name="Submit", button_type="primary")
+
+progress_message = pn.Row(
+    pn.indicators.LoadingSpinner(
+        value=True, width=25, height=25, align="center", margin=(5, 0, 5, 10)
+    ),
+    pn.panel("Running classifier ...", margin=0),
+)
+
+def get_prediction(running):
+    if not running:
+        yield "Click Submit"
+        return
+
+    yield progress_message
+    prediction = classify(None)
+    yield f"It's a {prediction}"
+
+pn.Column(run, pn.bind(get_prediction, run)).servable()
+```
+
+### Exercise: Disable the Button
+
+Update the code to `disable` the `Button` while the classification is running.
+
+:::::{dropdown}
+
+::::{tab-set}
+
+:::{tab-item} `.disabled`
+:sync: disabled
+
+```{pyodide}
+import panel as pn
+from time import sleep
+import random
+
+pn.extension()
+
+OPTIONS = ["Wind Turbine", "Solar Panel", "Battery Storage"]
+
+def classify(image):
+    sleep(2)
+    return random.choice(OPTIONS)
+
+
+run = pn.widgets.Button(name="Submit", button_type="primary")
+
+progress_message = pn.Row(
+    pn.indicators.LoadingSpinner(
+        value=True, width=25, height=25, align="center", margin=(5, 0, 5, 10)
+    ),
+    pn.panel("Running classifier ...", margin=0),
+)
+
+def get_prediction(running):
+    if not running:
+        yield "Click Submit"
+        return
+
+    run.disabled=True
+
+    yield progress_message
+    prediction = classify(None)
+    yield f"It's a {prediction}"
+
+    run.disabled=False
+
+pn.Column(run, pn.bind(get_prediction, run)).servable()
+```
+
+:::
+
+:::{tab-item} `.update`
+:sync: disabled
+
+```{pyodide}
+import panel as pn
+from time import sleep
+import random
+
+pn.extension()
+
+OPTIONS = ["Wind Turbine", "Solar Panel", "Battery Storage"]
+
+def classify(image):
+    sleep(2)
+    return random.choice(OPTIONS)
+
+
+run = pn.widgets.Button(name="Submit", button_type="primary")
+
+progress_message = pn.Row(
+    pn.indicators.LoadingSpinner(
+        value=True, width=25, height=25, align="center", margin=(5, 0, 5, 10)
+    ),
+    pn.panel("Running classifier ...", margin=0),
+)
+
+def get_prediction(running):
+    if not running:
+        yield "Click Submit"
+        return
+
+    with run.param.update(disabled=True):
+        yield progress_message
+        prediction = classify(None)
+        yield f"It's a {prediction}"
+
+pn.Column(run, pn.bind(get_prediction, run)).servable()
+```
+
+:::
+
+::::
+
+:::::
+
+### Use `.rx` state
+
+An alternative to a generator function is using *state* (`pn.rx`).
+
+Run the code below.
+
+```{pyodide}
+import random
```

Comment:
This example does not fully work due to https://github.com/holoviz/panel/issues/6302

Reply:
Fixed